sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala

```diff
@@ -418,4 +418,20 @@ class QueryExecutionErrorsSuite extends QueryTest
           |""".stripMargin)
     }
   }
+
+  test("FAILED_EXECUTE_UDF: execute user defined function") {
+    val e1 = intercept[SparkException] {
+      val words = Seq(("Jacek", 5), ("Agata", 5), ("Sweet", 6)).toDF("word", "index")
+      val luckyCharOfWord = udf { (word: String, index: Int) => {
+        word.substring(index, index + 1)
+      }}
+      words.select(luckyCharOfWord($"word", $"index")).collect()
+    }
+    assert(e1.getCause.isInstanceOf[SparkException])
+
+    val e2 = e1.getCause.asInstanceOf[SparkException]
+    assert(e2.getErrorClass === "FAILED_EXECUTE_UDF")
+    assert(e2.getMessage.matches("Failed to execute user defined function " +
+      "\\(QueryExecutionErrorsSuite\\$\\$Lambda\\$\\d+/\\d+: \\(string, int\\) => string\\)"))
```

Comment:
This seems to break Java 17 tests, @panbingkun and @MaxGekk . Please see the following community patch.
- https://github.com/apache/spark/pull/36270